File: misago/users/serializers/rank.py
from rest_framework import serializers

from ...core.utils import format_plaintext_for_html
from ..models import Rank

__all__ = ["RankSerializer"]


class RankSerializer(serializers.ModelSerializer):
    description = serializers.SerializerMethodField()
    url = serializers.SerializerMethodField()

    class Meta:
        model = Rank
        fields = [
            "id",
            "name",
            "slug",
            "description",
            "title",
            "css_class",
            "is_default",
            "is_tab",
            "url",
        ]

    def get_description(self, obj):
        if obj.description:
            return format_plaintext_for_html(obj.description)
        return ""

    def get_url(self, obj):
        return obj.get_absolute_url()

File: misago/frontend/misago/models/rank.js
(function (Misago) {
  'use strict';

  var Rank = function(data) {
    this.id = data.id;

    this.name = data.name;
    this.slug = data.slug;

    this.description = data.description;

    this.title = data.title;
    this.css_class = data.css_class;

    this.is_tab = data.is_tab;
  };

  Misago.addService('model:rank', function(_) {
    _.models.add('rank', {
      class: Rank
    });
  },
  {
    after: 'models'
  });
} (Misago.prototype));
